components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/internal/execution/PreUpdatePasswordRequestBuilder.java

```diff
@@ -86,6 +106,18 @@ private UserActionContext resolveUserActionContext(FlowContext flowContext)
         return (UserActionContext) userContext;
     }
 
+    private UniqueIDUserStoreManager resolveUserStoreManager(FlowContext flowContext)
+            throws ActionExecutionRequestBuilderException {
+
+        Object userStoreManager = flowContext.getContextData().get(PreUpdatePasswordActionConstants.USER_STORE_MANAGER);
+        if (!(userStoreManager instanceof UniqueIDUserStoreManager)) {
+            throw new ActionExecutionRequestBuilderException(
+                    "User store manager is not an instance of UniqueIDUserStoreManager for tenant: " +
+                            IdentityContext.getThreadLocalIdentityContext().getTenantDomain());
+        }
+        return (UniqueIDUserStoreManager) userStoreManager;
+    }
+
```

Comment:
Well this is not a good practice to follow.
UserStoreManager is a service why not retrieve it from realm service ?

Context commonly means properties, variables an models with related to the context. Not services

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6937/commits/ce0c5c5b7561381d8c8239a099c0e49ed20eed63